🐞 Vulnerability issue creator fails when Maven report does not contain "vulnerable" entry

# Summary
See https://github.com/exasol/extension-manager/actions/runs/6844839090/job/18609203687

```
Run tbx security cve convert maven < input | tee cves.jsonl
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/exasol/t │
│ oolbox/tools/security.py:187 in convert                                      │
│                                                                              │
│   184 │                                                                      │
│   185 │   actions = {Format.Maven: _maven}                                   │
│   186 │   action = actions[format]                                           │
│ ❱ 187 │   action(input_file)                                                 │
│   188                                                                        │
│   189                                                                        │
│   190 class Filter(str, Enum):                                               │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │     _maven = <function convert.<locals>._maven at 0x7fccfd90f600>        │ │
│ │     action = <function convert.<locals>._maven at 0x7fccfd90f600>        │ │
│ │    actions = {                                                           │ │
│ │              │   <Format.Maven: 'maven'>: <function                      │ │
│ │              convert.<locals>._maven at 0x7fccfd90f600>                  │ │
│ │              }                                                           │ │
│ │     format = <Format.Maven: 'maven'>                                     │ │
│ │ input_file = <_io.TextIOWrapper name='<stdin>' encoding='utf-8'>         │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.11.6/x64/lib/python3.11/site-packages/exasol/t │
│ oolbox/tools/security.py:181 in _maven                                       │
│                                                                              │
│   178 │                                                                      │
│   179 │   def _maven(infile):                                                │
│   180 │   │   issues = from_maven(infile.read())                             │
│ ❱ 181 │   │   for issue in _issues_as_json_str(issues):                      │
│   182 │   │   │   stdout(issue)                                              │
│   183 │   │   raise typer.Exit(code=0)                                       │
│   184                                                                        │
│                                                                              │
│ ╭─────────────────────────── locals ───────────────────────────╮             │
│ │ infile = <_io.TextIOWrapper name='<stdin>' encoding='utf-8'> │             │
│ │ issues = <generator object from_maven at 0x7fccfd940040>     │             │
│ │          {                                                               │ │
│ │          │   │   │   'coordinates':                                      │ │
│ │          'pkg:maven/com.fasterxml.jackson.core/jackson-annotations@2.15… │ │
│ │          │   │   │   'description': 'Core annotations used for value     │ │
│ │          types, used by Jackson databinding package.',                   │ │
│ │          │   │   │   'reference':                                        │ │
│ │          '[https://ossindex.sonatype.org/component/pkg:maven/com.fasterx…](https://ossindex.sonatype.org/component/pkg:maven/com.fasterx%E2%80%A6) │ │
│ │          │   │   },                                                      │ │
│ │          │   │   ... +96                                                 │ │
│ │          │   },                                                          │ │
│ │          │   'excludedVulnerabilities': [                                │ │
│ │          │   │   {                                                       │ │
│ │          │   │   │   'id': 'CVE-2023-4586',                              │ │
│ │          │   │   │   'displayName': 'CVE-2023-4586',                     │ │
│ │          │   │   │   'title': "[CVE-2023-4586] CWE-300: Channel          │ │
│ │          Accessible by Non-Endpoint ('Man-in-the-Middle'"+1,             │ │
│ │          │   │   │   'description': 'netty-handler - Improper            │ │
│ │          Certificate Validation\n\nThe product does not adequately'+230, │ │
│ │          │   │   │   'cvssScore': 6.5,                                   │ │
│ │          │   │   │   'cvssVector':                                       │ │
│ │          'CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:L/A:N',                 │ │
│ │          │   │   │   'cwe': 'CWE-300',                                   │ │
│ │          │   │   │   'cve': 'CVE-2023-4586',                             │ │
│ │          │   │   │   'reference':                                        │ │
│ │          '[https://ossindex.sonatype.org/vulnerability/CVE-2023-4586?com…](https://ossindex.sonatype.org/vulnerability/CVE-2023-4586?com%E2%80%A6) │ │
│ │          │   │   │   'externalReferences': [                             │ │
│ │          │   │   │   │                                                   │ │
│ │          '[https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLPa…](https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLPa%E2%80%A6) │ │
│ │          │   │   │   │   'https://github.com/netty/netty/issues/8537',   │ │
│ │          │   │   │   │   'https://github.com/netty/netty/issues/9930',   │ │
│ │          │   │   │   │                                                   │ │
│ │          '[https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html…](https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html%E2%80%A6) │ │
│ │          │   │   │   ]                                                   │ │
│ │          │   │   },                                                      │ │
│ │          │   │   {                                                       │ │
│ │          │   │   │   'id': 'CVE-2020-36641',                             │ │
│ │          │   │   │   'displayName': 'CVE-2020-36641',                    │ │
│ │          │   │   │   'title': '[CVE-2020-36641] CWE-611: Improper        │ │
│ │          Restriction of XML External Entity Reference '+7,               │ │
│ │          │   │   │   'description': 'A vulnerability classified as       │ │
│ │          problematic was found in gturri aXMLRPC up to 1.12'+586,        │ │
│ │          │   │   │   'cvssScore': 9.8,                                   │ │
│ │          │   │   │   'cvssVector':                                       │ │
│ │          'CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H',                 │ │
│ │          │   │   │   'cwe': 'CWE-611',                                   │ │
│ │          │   │   │   'cve': 'CVE-2020-36641',                            │ │
│ │          │   │   │   'reference':                                        │ │
│ │          '[https://ossindex.sonatype.org/vulnerability/CVE-2020-36641?co…](https://ossindex.sonatype.org/vulnerability/CVE-2020-36641?co%E2%80%A6) │ │
│ │          │   │   │   'externalReferences': [                             │ │
│ │          │   │   │   │                                                   │ │
│ │          '[http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2020-3664…](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2020-3664%E2%80%A6) │ │
│ │          │   │   │   │   'https://www.tenable.com/cve/CVE-2020-36641'    │ │
│ │          │   │   │   ]                                                   │ │
│ │          │   │   }                                                       │ │
│ │          │   ]                                                           │ │
│ │          }                                                               │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────╯
KeyError: 'vulnerable'
Error: Process completed with exit code 1.
```

--- a/doc/changelog.rst
+++ b/doc/changelog.rst
@@ -6,6 +6,10 @@
 Unreleased
 ==========
 
+🐞 Fixed
+--------
+* Fix failing vulnerability issue creator when Maven report does not contain "vulnerable" entry
+
 🔧 Changed
 ----------
 

--- a/exasol/toolbox/tools/security.py
+++ b/exasol/toolbox/tools/security.py
@@ -87,18 +87,17 @@ def gh_security_issues() -> Generator[Tuple[str, str], None, None]:
 def from_maven(report: str) -> Iterable[Issue]:
     # Note: Consider adding warnings if there is the same cve with multiple coordinates
     report = json.loads(report)
-    if "vulnerable" in report:
-        dependencies = report["vulnerable"]  # type: ignore
-        for _, dependency in dependencies.items():  # type: ignore
-            for v in dependency["vulnerabilities"]:  # type: ignore
-                references = [v["reference"]] + v["externalReferences"]
-                yield Issue(
-                    cve=v["cve"],
-                    cwe=v["cwe"],
-                    description=v["description"],
-                    coordinates=dependency["coordinates"],
-                    references=tuple(references),
-                )
+    dependencies = report.get("vulnerable", {})  # type: ignore
+    for _, dependency in dependencies.items():  # type: ignore
+        for v in dependency["vulnerabilities"]:  # type: ignore
+            references = [v["reference"]] + v["externalReferences"]
+            yield Issue(
+                cve=v["cve"],
+                cwe=v["cwe"],
+                description=v["description"],
+                coordinates=dependency["coordinates"],
+                references=tuple(references),
+            )
 
 
 def security_issue_title(issue: Issue) -> str:

--- a/exasol/toolbox/tools/security.py
+++ b/exasol/toolbox/tools/security.py
@@ -87,17 +87,18 @@ def gh_security_issues() -> Generator[Tuple[str, str], None, None]:
 def from_maven(report: str) -> Iterable[Issue]:
     # Note: Consider adding warnings if there is the same cve with multiple coordinates
     report = json.loads(report)
-    dependencies = report["vulnerable"]  # type: ignore
-    for _, dependency in dependencies.items():  # type: ignore
-        for v in dependency["vulnerabilities"]:  # type: ignore
-            references = [v["reference"]] + v["externalReferences"]
-            yield Issue(
-                cve=v["cve"],
-                cwe=v["cwe"],
-                description=v["description"],
-                coordinates=dependency["coordinates"],
-                references=tuple(references),
-            )
+    if "vulnerable" in report:
+        dependencies = report["vulnerable"]  # type: ignore
+        for _, dependency in dependencies.items():  # type: ignore
+            for v in dependency["vulnerabilities"]:  # type: ignore
+                references = [v["reference"]] + v["externalReferences"]
+                yield Issue(
+                    cve=v["cve"],
+                    cwe=v["cwe"],
+                    description=v["description"],
+                    coordinates=dependency["coordinates"],
+                    references=tuple(references),
+                )
 
 
 def security_issue_title(issue: Issue) -> str:

--- a/test/unit/security_test.py
+++ b/test/unit/security_test.py
@@ -354,3 +354,8 @@ def test_convert_maven_input(maven_report):  # pylint: disable=redefined-outer-n
     }
     actual = set(security.from_maven(maven_report))
     assert actual == expected
+
+
+def test_convert_maven_input_no_vulnerable():  # pylint: disable=redefined-outer-name
+    actual = set(security.from_maven("{}"))
+    assert len(actual) == 0

